### ISSUE TITLE
Dealing with DNA extraction and sequencing negative controls together

### ISSUE THREAD

**Issue by shreyaskumbhare** [benjjneb/DecontamManuscript#1]
@benjjneb Hi! I am dealing with data with two sets of negative controls: DNA extraction reagent control and sequencing reagent control. As the DNA extraction was done in 4 batches and the sequencing in 2 batches, I am unable to decide the strategy to carry out the decontamination. I was thinking of identifying the contaminants in two steps (first with DNA extraction controls and then with the sequencing run controls). however, I wish to know if I can do this together? Meaning, defining my samples with DNA extraction batch and sequencing run in my metadata/sample data file and then carrying out the decontamination together.

I would like to know your thoughts on this, thanks!

**Comment by benjjneb** [comment]
> I was thinking of identifying the contaminants in two steps (first with DNA extraction controls and then with the sequencing run controls).

That seems like a decent strategy to me. You could run `isContaminant` twice, with `batch=extraction` and `batch=run`, and then consider the contaminants identified either way.

> I wish to know if I can do this together? Meaning, defining my samples with DNA extraction batch and sequencing run in my metadata/sample data file and then carrying out the decontamination together.

The function itself only supports a single batch factor. You could create some compound factor, like extraction x run, but I think just considering each processing batch separately (as you suggested above) would work.

**Comment by shreyaskumbhare** [comment]
@benjjneb Thanks, that helped! I did it in two steps and worked out well.

**Comment by shreyaskumbhare** [comment]
Hi @benjjneb I tried out the decontam using the isContam function as discussed above in two steps. However, I was trying to play around with the threshold value. I tried using the default (threshold=0.1) and a more stringent (threshold=0.5), and as expected I could see a drastic difference in the outputs of two. However, my major concern is using a more aggressive approach I am loosing out on an average of 90% of the reads in my samples Vs. 6% using the default threshold.

Following are the outputs:

1. With a 0.5 threshold:
<img width="761" alt="Screen Shot 2019-11-06 at 4 49 13 PM" src="https://user-images.githubusercontent.com/40797916/68344790-834a8100-00b5-11ea-830b-908c695ad8ab.png">

2. With a 0.1 threshold:
<img width="695" alt="Screen Shot 2019-11-06 at 4 50 04 PM" src="https://user-images.githubusercontent.com/40797916/68344811-9a896e80-00b5-11ea-8109-0f2a717559a7.png">

I would love to hear your thoughts on this, thanks!

**Comment by benjjneb** [comment]
The power of prevalence-based contaminant identification is limited by the number of negative contorls you have. You have 4, which isn't that many really. I tend to be more cautious removing contaminants here, and use a more conservative threshold like 0.1, rather than the more aggressive threshold of 0.5.

That said, if you know you are working in a (very) low biomass environment, aggression could be warranted, so this is context dependent.

**Comment by shreyaskumbhare** [comment]
oh ok, I got your point. Unfortunately, for now, I have just one DNA extraction reagent control per batch which I used in the first step and then I had a sequencing control per run which was used in second step for decontamination. I think in this case using a threshold=0.1 would be appropriate as I don't have enough controls. These are fecal samples from infants, so I won't really say those were "very low" biomass samples. However, I will keep this in mind for future studies to have multiple controls per extraction and sequencing batch; are there any recommendations/suggestions for the number of controls that won't be considered as "underpowered"? Thanks!

**Comment by benjjneb** [comment]
> are there any recommendations/suggestions for the number of controls that won't be considered as "underpowered"?

In the manuscript we suggest 5-6 based on some simulation analyses. Classification accuracy will only go up with more though.

Also, the frequency method does not require negative controls at all, just that the DNA quantitation data was recorded for each sample.

**Comment by shreyaskumbhare** [comment]
Hi @benjjneb thank you for your response. I am convinced that more replicates will have better accuracy in identifying true contaminants. I tried all possible combinations (including the threshold parameter) with my dataset and then finally decided to use all the controls (DNA extraction controls) together, instead of using a batch factor. Because in my case I had only one control per DNA extraction batch. In this way, I know that I will lose the batch effect, however, using them together (as replicates) will at least have better accuracy and I have tested that personally on my dataset.

But having said that, I will surely keep in mind to have at least triplicates per DNA extraction batch and then set of triplicate for my library controls for sequencing. Thank you for your responses!

Cheers,
Shreyas
